internal/server/create_document.go

```diff
@@ -97,46 +101,80 @@ func CreateDocument(client CreateDocumentClient, tmpl template.Template) Handler
 			data.DocumentTemplateTypes = translateDocumentData(data.DocumentTemplates, data.DocumentTemplateRefData)
 
 			templateId := r.FormValue("templateId")
+			hasSelectedSubmitTemplate := r.FormValue("hasSelectedSubmitTemplate")
+
 			if templateId != "" {
 				for _, dt := range data.DocumentTemplates {
 					if dt.TemplateId == templateId {
 						data.TemplateSelected = dt
 						break
 					}
 				}
+			} else if hasSelectedSubmitTemplate == "true" {
+				w.WriteHeader(http.StatusBadRequest)
+				data.Error = sirius.ValidationError{
+					Field: sirius.FieldErrors{
+						"templateId": {"reason": "Please select a document template to continue"},
+					},
+				}
 			}
+
 			if data.TemplateSelected.TemplateId != "" {
 				data.DocumentInsertTypes = translateInsertData(data.TemplateSelected.Inserts, data.DocumentTemplateRefData)
+				data.DocumentInsertKeys = getSortedInsertKeys(data.TemplateSelected.Inserts)
 			}
 
 			hasViewedInsertPage := r.FormValue("hasViewedInserts")
-			if hasViewedInsertPage == "true" {
+			hasNoInsertsToSelect := data.TemplateSelected.TemplateId != "" && len(data.DocumentInsertTypes) == 0
+			uniqueInserts := removeDuplicateStr(r.Form["insert"])
```

Comment:
Had to handle remove duplicate inserts as an insert can appear in multiple tabs and all ids are posted with the form